In [93]:
import pandas as pd

In [94]:
allergies = pd.read_csv('allergies.csv')
conditions = pd.read_csv('conditions.csv')
encounters = pd.read_csv('encounters.csv')
patients = pd.read_csv('patients.csv')
medications = pd.read_csv("medications.csv")
observations = pd.read_csv("observations.csv")
patients = pd.read_csv("patients.csv")
procedures = pd.read_csv("procedures.csv")


In [3]:
allergies.columns

Index(['START', 'STOP', 'PATIENT', 'ENCOUNTER', 'CODE', 'DESCRIPTION'], dtype='object')

In [4]:
conditions.columns

Index(['START', 'STOP', 'PATIENT', 'ENCOUNTER', 'CODE', 'DESCRIPTION'], dtype='object')

In [5]:
encounters.columns

Index(['Id', 'START', 'STOP', 'PATIENT', 'PROVIDER', 'PAYER', 'ENCOUNTERCLASS',
       'CODE', 'DESCRIPTION', 'BASE_ENCOUNTER_COST', 'TOTAL_CLAIM_COST',
       'PAYER_COVERAGE', 'REASONCODE', 'REASONDESCRIPTION'],
      dtype='object')

In [12]:
patients.columns

Index(['Id', 'BIRTHDATE', 'DEATHDATE', 'SSN', 'DRIVERS', 'PASSPORT', 'PREFIX',
       'FIRST', 'LAST', 'SUFFIX', 'MAIDEN', 'MARITAL', 'RACE', 'ETHNICITY',
       'GENDER', 'BIRTHPLACE', 'ADDRESS', 'CITY', 'STATE', 'COUNTY', 'ZIP',
       'LAT', 'LON', 'HEALTHCARE_EXPENSES', 'HEALTHCARE_COVERAGE'],
      dtype='object')

In [13]:
len(patients['Id'].unique())

1180

### There are total 1180 unique patients in this dataset. ID is the Unique Identifier of the patient.

In [16]:
enc_array = encounters['PATIENT'].unique()

In [22]:
count = 0
for i in enc_array:
    if i in patients['Id'].unique():
        count +=1

In [23]:
count

1180

### PATIENT identifier is same as ID of a patient.

### Idea 1: Encounter is the ID given to unique identifyt the patient visits to the hospital and let's see number of encounters for a single patient.

In [24]:
enc_ids = encounters['Id'].unique()
len(enc_ids)

46868

In [25]:
count = 0
for i in enc_array:
    if i in enc_ids:
        count +=1
print(count)

0


In [32]:
encounters.head()

Id                 START  \
0  1c97b103-d387-4804-857d-5e5b719edb65  2011-06-12T10:25:43Z   
1  64efa414-238b-417f-8e8b-10f0949b6f06  2014-06-15T10:25:43Z   
2  0b4374e8-838d-4289-b491-f44d51360790  2016-04-14T10:25:43Z   
3  b86c6047-c25b-4b9c-90df-5712852902aa  2017-06-18T10:25:43Z   
4  61f6867a-9e58-48e6-87d0-6d9495a81a79  1995-08-18T11:57:51Z   

                   STOP                               PATIENT  \
0  2011-06-12T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
1  2014-06-15T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
2  2016-04-14T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
3  2017-06-18T10:55:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
4  1995-08-18T12:12:51Z  3603cd65-53a3-424b-bc82-76204326510d   

                               PROVIDER                                 PAYER  \
0  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   
1  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   
2  4861d01f-019c-3dac-a153-8334e50919f9  d47b3510-2895-3b70-9897-342d681c769d   
3  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   
4  dd43b798-6c99-3cdb-bd19-65f1cc2a370d  d47b3510-2895-3b70-9897-342d681c769d   

  ENCOUNTERCLASS       CODE                                 DESCRIPTION  \
0       wellness  162673000  General examination of patient (procedure)   
1       wellness  162673000  General examination of patient (procedure)   
2     ambulatory  185345009                       Encounter for symptom   
3       wellness  162673000  General examination of patient (procedure)   
4       wellness  162673000  General examination of patient (procedure)   

   BASE_ENCOUNTER_COST  TOTAL_CLAIM_COST  PAYER_COVERAGE   REASONCODE  \
0               129.16            129.16          129.16          NaN   
1               129.16            129.16          129.16          NaN   
2               129.16            129.16           49.16  444814009.0   
3               129.16            129.16          129.16          NaN   
4               129.16            129.16           49.16          NaN   

            REASONDESCRIPTION  
0                         NaN  
1                         NaN  
2  Viral sinusitis (disorder)  
3                         NaN  
4                         NaN

In [38]:
single_patient = encounters[encounters['PATIENT'] == 'd171d808-1f31-4ad3-aba5-e03a2fa921c7']

In [41]:
single_patient.head()

Id                 START  \
0  1c97b103-d387-4804-857d-5e5b719edb65  2011-06-12T10:25:43Z   
1  64efa414-238b-417f-8e8b-10f0949b6f06  2014-06-15T10:25:43Z   
2  0b4374e8-838d-4289-b491-f44d51360790  2016-04-14T10:25:43Z   
3  b86c6047-c25b-4b9c-90df-5712852902aa  2017-06-18T10:25:43Z   

                   STOP                               PATIENT  \
0  2011-06-12T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
1  2014-06-15T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
2  2016-04-14T10:40:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   
3  2017-06-18T10:55:43Z  d171d808-1f31-4ad3-aba5-e03a2fa921c7   

                               PROVIDER                                 PAYER  \
0  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   
1  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   
2  4861d01f-019c-3dac-a153-8334e50919f9  d47b3510-2895-3b70-9897-342d681c769d   
3  6edd4924-266f-3c01-b2ca-a1345cfb12e3  d47b3510-2895-3b70-9897-342d681c769d   

  ENCOUNTERCLASS       CODE                                 DESCRIPTION  \
0       wellness  162673000  General examination of patient (procedure)   
1       wellness  162673000  General examination of patient (procedure)   
2     ambulatory  185345009                       Encounter for symptom   
3       wellness  162673000  General examination of patient (procedure)   

   BASE_ENCOUNTER_COST  TOTAL_CLAIM_COST  PAYER_COVERAGE   REASONCODE  \
0               129.16            129.16          129.16          NaN   
1               129.16            129.16          129.16          NaN   
2               129.16            129.16           49.16  444814009.0   
3               129.16            129.16          129.16          NaN   

            REASONDESCRIPTION  
0                         NaN  
1                         NaN  
2  Viral sinusitis (disorder)  
3                         NaN

In [42]:
single_patient.shape

(4, 14)

### Task 1: Find two patients who have more than 100 records or encounter records.

In [44]:
patients = encounters['PATIENT'].unique()
patient_more_than_100 = []
for i in patients:
    patient_records = encounters[encounters['PATIENT'] == i] 
    if patient_records.shape[0] > 100 :
        patient_more_than_100.append(i)

        
        


In [45]:
patient_more_than_100

['ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7',
 '2c57a897-8381-44a6-920f-e074fa6f74cf',
 'bd2a1c40-c248-4ee4-a2d1-bfa8f0cfd0f1',
 'c85e59cb-50b9-4ae5-9961-12a5008e640a',
 'cc257fb6-c91f-46b6-80cb-d9a9a3fc9dd7',
 '9da0dcfc-05e3-4e8e-95ff-b04b56f748be',
 'ae6669f0-0200-4f8d-83cc-d977041acad5',
 'a82ffcdd-11db-47d9-ab30-9e5a2c51194a',
 '6db58d94-662c-4d07-bf79-2129ca4b8f14',
 '4a9a97d7-5fa5-40ed-945d-72f33872c632',
 '8027e27d-405f-490c-a946-91d1b40c8fd0',
 'b3e1def9-d4cf-4c54-a15a-38e936f06235',
 '24b1778f-54bf-4d39-b6b7-6cc780bd3f24',
 '1b0fa3ec-629f-480c-85c4-61c9ec57796c',
 '3bf1dbdb-6332-4bf3-949f-77c9c2c5b9f8',
 'c71e1837-f1fb-4ac4-a05b-216cccfb0a9d',
 '369b9c8d-aab5-4d69-bdbc-fbf4a052a8a0',
 'bd58d43b-a597-409c-82b1-89db75ca275a',
 'ff76c6f0-d0da-438d-bce4-0c4e88fbc752',
 '25195d87-a884-436d-b6bf-d97191acf5c3',
 '3a24d9f2-1389-4e32-b002-d71762ef4435',
 '28ea39f3-a07b-457b-b12d-e35c8bd2d1dd',
 '0979f4fe-08c5-414e-ba1c-6ccf852bcce4',
 'e978a995-4b48-4c1b-8bd3-629b98b6b6b0',
 '2f88c354-3ca7-

In [85]:
patient_1 = encounters[encounters['PATIENT'] == 'ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7']
patient_2 = encounters[encounters['PATIENT'] == '2c57a897-8381-44a6-920f-e074fa6f74cf']

In [86]:
patient_1

Id                 START  \
2454  fcca7970-f7fe-4eb0-bf93-f26ab9d219af  1950-11-04T01:22:05Z   
2455  807c459f-83a5-4b37-8e04-7610777efb5f  1950-11-04T01:22:05Z   
2456  b852ca22-9018-4db2-a51f-848070b9a187  1951-02-21T01:22:05Z   
2457  a0d09b07-bb32-46f4-a35e-0b457e626e27  1952-05-02T01:22:05Z   
2458  ad8a65dd-4b70-4308-a2e9-0af4ca29484e  1959-11-28T01:22:05Z   
2459  67746bca-e630-4706-a09c-1c9e2c203dfd  1962-12-01T01:22:05Z   
2460  a15e876d-9cfc-4090-ba70-81a7a7a39da5  1965-12-04T01:22:05Z   
2461  0ff35128-cd8f-4468-9ddc-462cfa70e386  1971-09-25T01:22:05Z   
2462  5e56baf4-a701-4a11-8f70-8c77f482e9c2  1973-09-29T01:22:05Z   
2463  99e5210d-fb88-446e-8762-180c1bb27c41  1975-10-04T01:22:05Z   
2464  88098823-3aea-4ea8-8d43-365467a884f1  1977-10-08T01:22:05Z   
2465  734f62c4-6aa6-4a67-8530-054b8a103ad1  1977-12-24T01:22:05Z   
2466  de378ef2-5156-442d-ae15-2f09e43539c8  1983-09-24T01:22:05Z   
2467  387c290e-f024-4967-ac89-54ebc40835e9  1983-10-01T01:22:05Z   
2468  91d34cec-6afe-4719-94c5-e73d2371b8a5  1991-09-07T01:22:05Z   
2469  2200ca7a-24a3-4ba4-b5e3-ec3b8cc698fb  1992-05-09T01:22:05Z   
2470  d03663d6-b698-4a7e-8051-b3ac09baf897  1994-01-29T01:22:05Z   
2471  4f03a155-4be3-406b-8190-4236a7faebbc  1995-12-09T01:22:05Z   
2472  33cd324d-aa2f-45c5-acc6-1373821f7e3f  1997-01-11T01:22:05Z   
2473  42680660-0405-47f0-a3d2-3859bc2b000a  2009-10-03T01:22:05Z   
2474  7b3e6441-904f-4fac-9d47-0f9c7e99ba4e  2009-10-10T01:22:05Z   
2475  f4a7881a-6591-4d09-a39d-7d67b708416c  2009-10-31T01:22:05Z   
2476  0839d6f8-467a-4550-8289-ed23240f963b  2009-11-07T01:22:05Z   
2477  7df80c9a-4ad0-4c4b-a9d6-7efbbfdbffa8  2009-11-14T01:22:05Z   
2478  47925c79-3ea0-49d6-b669-30b7ce814254  2009-12-05T01:22:05Z   
2479  35eb403a-f824-4b62-87d3-26b511288d4e  2009-12-12T01:22:05Z   
2480  e6611738-a340-4a03-8a81-5b2d0837ce2d  2009-12-19T01:22:05Z   
2481  d3d7181f-e63e-4465-9e86-566df69f90bf  2010-01-02T01:22:05Z   
2482  258d0fbe-f99a-45a1-81c8-28a89d17757e  2010-01-21T01:22:05Z   
2483  1df7b724-8dd9-43e6-aa3e-c77d3cc3ee85  2010-01-30T01:22:05Z   
...                                    ...                   ...   
2885  6798d426-95db-436d-a14f-070a5e7534ed  2019-02-16T01:22:05Z   
2886  95d6b27b-429a-4a4b-ab90-7292a1b0c7fd  2019-02-23T01:22:05Z   
2887  1b08b1ce-2f06-4af0-84d8-5b6e4eb40919  2019-03-02T01:22:05Z   
2888  e29e2882-fce4-40e8-9515-9c91f36a8aae  2019-03-09T01:22:05Z   
2889  e3e7ac42-195e-4983-8d7b-fbe673d9e54f  2019-03-16T01:22:05Z   
2890  734ee876-2fd7-4f63-aded-ae8c7f9720a2  2019-03-23T01:22:05Z   
2891  7673b818-6b8e-468b-9870-3192fcf52950  2019-03-30T01:22:05Z   
2892  db5460d2-c5b4-4168-b50c-ca24f3609b78  2019-04-06T01:22:05Z   
2893  43a50475-a23d-4aa6-bc30-71db6d2a67b4  2019-04-13T01:22:05Z   
2894  4a62384f-ec23-4448-8670-98c4f12fb61b  2019-04-20T01:22:05Z   
2895  67525d8a-3b83-4f34-afaf-c6954c172ffa  2019-04-27T01:22:05Z   
2896  1804b341-a6ce-4abd-b485-f32f5113ced4  2019-05-04T01:22:05Z   
2897  b57c39dc-702d-4186-bf71-9578ac2a5ec9  2019-05-11T01:22:05Z   
2898  ba1541d1-b127-4e45-9366-a944cc5298db  2019-05-18T01:22:05Z   
2899  43f5ecdb-afa2-49d4-8067-be85d609b19f  2019-05-25T01:22:05Z   
2900  0802b19a-ead6-4a61-849a-1a6ce9974ebf  2019-06-01T01:22:05Z   
2901  df56c5ce-a2a4-41ad-b934-e01b9c02a534  2019-06-08T01:22:05Z   
2902  e3f24ca1-269c-497a-b4df-e85bb6dbe774  2019-06-15T01:22:05Z   
2903  7e324007-c9d3-437e-a0a9-3445c0d02139  2019-06-22T01:22:05Z   
2904  53bf3339-51cd-4157-a465-dbe0715c3843  2019-06-29T01:22:05Z   
2905  ed04d165-42c8-48ea-99d3-ca27aaf7fb91  2019-07-06T01:22:05Z   
2906  6fc4435f-cc09-429e-9e82-fc6fe102a8fa  2019-07-13T01:22:05Z   
2907  c868b2fa-4740-4272-b3ae-891ef25d5efd  2019-07-20T01:22:05Z   
2908  233818f6-29aa-4964-b3cf-1c3cf9ec86c3  2019-07-27T01:22:05Z   
2909  86636fb6-7f0e-4943-8feb-63d78d1deffa  2019-08-03T01:22:05Z   
2910  5256233c-c1c1-47b3-8141-e5a24cdefb92  2019-08-10T01:22:05Z   
2911  1452c200-c9d3-42e2-b35b-04551696f6ed  2019-08-17T01:22:05Z   
2912  b25614ac-d8e4-46d7-a40

In [87]:
patient_2

Id                 START  \
3265  bc05b610-9a2e-404f-8225-7b09d7a7f9c6  1946-07-29T10:52:17Z   
3266  efd8af52-f35b-47cf-994b-6a44c7f479f2  1947-01-13T10:52:17Z   
3267  99507900-490e-4879-a3b6-e6e8b7e474e3  1949-08-15T10:52:17Z   
3268  29ef6631-a349-4c6d-b45f-18d02a89344e  1980-08-18T10:52:17Z   
3269  993d53e9-435c-4933-8855-ca4d96c01569  1984-08-27T10:52:17Z   
3270  01f462a2-4925-4102-b2ba-7fbac893b099  1986-09-01T10:52:17Z   
3271  0c72e5de-506d-4341-97b9-ad32cddd3413  1988-09-05T10:52:17Z   
3272  eacca7f4-1081-4a8e-b351-b7f4292a71bd  1999-09-27T10:52:17Z   
3273  7ae51be8-300f-4326-9406-c146edc8aca5  1999-10-04T10:52:17Z   
3274  d92702ca-372c-4157-b33f-e228ddb741d4  1999-11-29T10:52:17Z   
3275  43009d1b-2c72-41cb-99a0-53e74fedf51f  1999-12-06T10:52:17Z   
3276  fa9b8d1e-7e3e-4f6d-89cd-82ab9e2785ac  1999-12-13T10:52:17Z   
3277  f44978c7-cacb-4cbe-a02c-566337197006  1999-12-27T10:52:17Z   
3278  55af9d22-41ff-4670-ac34-1129e347a36e  2000-01-24T10:52:17Z   
3279  c3ba49d7-1204-4e90-b5d5-179e9d3bba0d  2000-02-21T10:52:17Z   
3280  a19c3f92-2ebd-4cfe-a4de-d143b38b4956  2000-04-24T10:52:17Z   
3281  48113562-b1a1-4d84-8894-c17390c012b4  2000-05-01T10:52:17Z   
3282  1fe29272-e5e4-41ba-8d7e-e74812505339  2000-05-22T10:52:17Z   
3283  36992efb-f033-4930-b109-d8449b403919  2000-06-26T10:52:17Z   
3284  cd7d2d18-fee5-45de-af8b-b0a438b9523d  2000-07-03T10:52:17Z   
3285  03c2dc0d-6895-42da-86e7-7934820996e0  2000-07-24T10:52:17Z   
3286  c139baa9-13c7-4aca-88b8-faa14e558527  2000-07-31T10:52:17Z   
3287  6d8b58bf-cae9-4769-ab4b-e33fbf90e9ad  2000-08-10T10:52:17Z   
3288  0edabdaf-1acd-47a5-a467-936384059154  2000-08-21T10:52:17Z   
3289  c8dcce49-c89a-40e4-aa37-9fecaae80160  2000-08-28T10:52:17Z   
3290  2b796eab-5259-47b3-9b8c-48fab57b5d4d  2000-09-04T10:52:17Z   
3291  6383dc36-1078-480f-a770-9dde26be2fd2  2000-09-11T10:52:17Z   
3292  79f0b7ea-3741-4633-804a-03b9dd4c9b03  2000-09-07T10:52:17Z   
3293  d2e1ed2b-9314-4f6a-bf8f-935b951260a3  2000-09-18T10:52:17Z   
3294  45a178c4-fe1d-4502-9619-3998759ad8f6  2000-09-25T10:52:17Z   
...                                    ...                   ...   
3677  2ecb4223-9a14-4650-9dc8-7c58cabf97f4  2009-07-09T10:52:17Z   
3678  f99981d7-dd37-4ee4-9b3e-0c039bd7b887  2009-07-12T10:52:17Z   
3679  e99a7daf-1901-4564-86ae-be2b11beff79  2009-07-15T10:52:17Z   
3680  da2707fd-825d-4ef6-a651-20f402fb8d1f  2009-07-18T10:52:17Z   
3681  de5ce0fc-357d-4a4d-92a0-81c3b61e2a6e  2009-07-21T10:52:17Z   
3682  1b58e85b-3a07-46b4-90d5-9089c973f309  2009-07-24T10:52:17Z   
3683  9ff65486-e405-4fbf-ac7d-85453dbda1f6  2009-07-27T10:52:17Z   
3684  0d0e881c-504d-4dbc-bed5-2a7f40012f64  2009-08-03T10:52:17Z   
3685  028e7a30-38ba-459e-af36-f6c911494e00  2009-07-30T10:52:17Z   
3686  20b0e448-895c-4e3f-a1bc-0ebac170afd3  2009-08-02T10:52:17Z   
3687  661ef9a9-dd27-4f01-b6fd-da4fb2fa4065  2009-07-29T10:52:17Z   
3688  ec3a82db-89b9-467c-a2d0-a00df65d05ae  2009-08-05T10:52:17Z   
3689  e58936cc-ebca-42a7-9cfd-1a16fd3c8926  2009-08-05T10:52:17Z   
3690  2af60500-5baf-42f4-b20a-13c5baebd514  2009-08-08T10:52:17Z   
3691  a021477d-4454-4ce0-8ea6-516cdb1a2d74  2009-08-11T10:52:17Z   
3692  a4d66b5e-5b05-4b24-83f3-a5e86d156f2f  2009-08-14T10:52:17Z   
3693  4581d48e-6f83-4334-af3d-3a316e668bae  2009-08-17T10:52:17Z   
3694  9984667e-a411-482b-a5b1-309e9d5fb56f  2009-08-20T10:52:17Z   
3695  418be66b-3f5c-44bd-9087-51085a95ba76  2009-08-23T10:52:17Z   
3696  aca5e061-4643-4114-a0fd-7b4da142b9ac  2009-08-26T10:52:17Z   
3697  d68c93f1-1a1e-4f75-94d8-98f71adc5103  2009-08-29T10:52:17Z   
3698  d43611af-70ad-48a9-8df5-124f1846868f  2009-09-07T10:52:17Z   
3699  71d5f2e6-d767-4685-8ebb-536ab6fb1867  2009-09-01T10:52:17Z   
3700  3d1fe08b-fc43-446d-8d8f-f54fc46e2dae  2009-09-04T10:52:17Z   
3701  c1c6cbe6-141c-49ce-8d33-09dfd8f5e6dc  2009-09-04T10:52:17Z   
3702  f52838f2-e8f2-40b0-9946-519f8136e88e  2009-09-07T10:52:17Z   
3703  ace6b356-8bc0-4d6b-b3bd-768faea591ef  2009-09-01T10:52:17Z   
3704  a80f57fe-8b9b-4f57-886

### Let's build the above dataframes with the needed information.

Dropping columns from BASE_ENCOUNTER_COST..

In [88]:
patient_1 = patient_1.drop(patient_1.iloc[:, 7:], axis = 1) 
patient_2 = patient_2.drop(patient_2.iloc[:, 7:], axis = 1) 

In [89]:
patient_1.head()

Id                 START  \
2454  fcca7970-f7fe-4eb0-bf93-f26ab9d219af  1950-11-04T01:22:05Z   
2455  807c459f-83a5-4b37-8e04-7610777efb5f  1950-11-04T01:22:05Z   
2456  b852ca22-9018-4db2-a51f-848070b9a187  1951-02-21T01:22:05Z   
2457  a0d09b07-bb32-46f4-a35e-0b457e626e27  1952-05-02T01:22:05Z   
2458  ad8a65dd-4b70-4308-a2e9-0af4ca29484e  1959-11-28T01:22:05Z   

                      STOP                               PATIENT  \
2454  1950-11-04T01:37:05Z  ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7   
2455  1950-11-04T01:52:05Z  ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7   
2456  1951-02-21T01:52:05Z  ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7   
2457  1952-05-02T01:37:05Z  ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7   
2458  1959-11-28T01:52:05Z  ad7f38c5-8a53-4a6c-9b6d-7ad0637164d7   

                                  PROVIDER  \
2454  b1ddf812-1fdd-3adf-b1d5-32cc8bd07ebb   
2455  b1ddf812-1fdd-3adf-b1d5-32cc8bd07ebb   
2456  b1ddf812-1fdd-3adf-b1d5-32cc8bd07ebb   
2457  b1ddf812-1fdd-3adf-b1d5-32cc8bd07ebb   
2458  df8d9ee7-fec4-30c7-b12c-f13b88916bca   

                                     PAYER ENCOUNTERCLASS  
2454  6e2f1a2d-27bd-3701-8d08-dae202c58632     ambulatory  
2455  6e2f1a2d-27bd-3701-8d08-dae202c58632     ambulatory  
2456  6e2f1a2d-27bd-3701-8d08-dae202c58632     ambulatory  
2457  6e2f1a2d-27bd-3701-8d08-dae202c58632     ambulatory  
2458  6e2f1a2d-27bd-3701-8d08-dae202c58632       wellness

### Adding Allergy Column

In [91]:
patient_1_encouter_id = patient_1['Id'].tolist()
patient_2_encouter_id = patient_2['Id'].tolist()
patient_1_id = patient_1['PATIENT'].tolist()
patient_2_id = patient_2['PATIENT'].tolist()


In [92]:
for i,j in zip(patient_1_id,patient_1_encouter_id):
    
    for index, row in allergies.iterrows():
        if row['PATIENT'] == i and row['ENCOUNTER'] == j:
            patient_1['Allergy'] = row['DESCRIPTION']
        else:
            patient_1['Allergy'] = "None"
            
for i,j in zip(patient_2_id,patient_2_encouter_id):
    
    for index, row in allergies.iterrows():
        if row['PATIENT'] == i and row['ENCOUNTER'] == j:
            patient_2['Allergy'] = row['DESCRIPTION']
        else:
            patient_2['Allergy'] = "None"
            
        

In [95]:
patient_1["Allergy"].value_counts()

None    461
Name: Allergy, dtype: int64

In [96]:
patient_2["Allergy"].value_counts()

None    407
Name: Allergy, dtype: int64